features/frameworks/mbed-coap/source/sn_coap_builder.c

```diff
@@ -361,6 +363,24 @@ uint16_t sn_coap_builder_calc_needed_packet_data_size_2(const sn_coap_hdr_s *src
     }
     return returned_byte_count;
 }
+
+/**
+ * \fn static bool sn_coap_builder_check_uint16_overflow(uint16_t param_a, uint16_t param_b))
+ *
+ * \brief Check that can param_a and param_b addition can be performed without overflow
+ *
+ * \param  param_a some uint16_t value
+ *
+ * \param  param_b some uint16_t value
+ *
+ * \return Returns true is would not overflow
+ */
+static bool sn_coap_builder_check_uint16_overflow(uint16_t param_a, uint16_t param_b)
+{
+    return param_a + param_b <= UINT16_MAX;
```

Comment:
This test only works if `int` is > 16-bits, so wouldn't work on MSP430.

     return (uint16_t) (param_a + param_b) >= param_a;

would work on anything.